dissect/target/plugins/os/windows/notifications.py

```diff
@@ -63,11 +198,149 @@ def __init__(self, target):
                 self.wpndb_files.append((user_details.user, wpndb_file))
 
             if appdb_file.exists():
-                self.appdb_files.append((user_details.user, appdb_file))
+                with appdb_file.open(mode="rb") as fp:
+                    chunk = c_appdb.Chunk(fp)
+                    if chunk.Header.Magic == APPDB_MAGIC and chunk.Header.Version == 3:
+                        self.appdb_files.append((user_details.user, appdb_file))
 
     def check_compatible(self) -> None:
         if not self.wpndb_files and not self.appdb_files:
-            raise UnsupportedPluginError("No wpndatabase.db or appdb.dat files found")
+            raise UnsupportedPluginError("No or incompatible wpndatabase.db or appdb.dat files found")
+
+    def _get_appdb_chunk_record(
+        self,
+        chunk: c_appdb.Chunk,
+        last_notification_date: Optional[datetime.datetime],
+        user: RecordDescriptor,
+    ) -> AppDBChunkRecord:
+        push_timestamp1 = None
+        if ts := chunk.PushDescriptor.Timestamp1:
+            push_timestamp1 = wintimestamp(ts)
+        push_timestamp2 = None
+        if ts := chunk.PushDescriptor.Timestamp1:
+            push_timestamp2 = wintimestamp(ts)
+        push_uri = chunk.PushDescriptor.Uri.split(b"\x00")[0]
+        push_uri = push_uri.decode("utf-8", errors="replace")
+
+        badge_ts = None
+        if ts := chunk.BadgeContent.Timestamp:
+            badge_ts = wintimestamp(ts)
+        badge_data = chunk.BadgeContent.Data.decode("utf-8", errors="replace")
+
+        return AppDBChunkRecord(
+            last_notification_date=last_notification_date,
+            next_notification_id=chunk.Header.NextNotificationId,
+            push_ts1=push_timestamp1,
+            push_ts2=push_timestamp2,
+            push_uri=push_uri,
+            badge_id=chunk.BadgeContent.Id,
+            badge_ts=badge_ts,
+            badge_data=badge_data,
+            _target=self.target,
+            _user=user,
+        )
+
+    def _get_appdb_tile_records(self, chunk: c_appdb.Chunk, user: RecordDescriptor) -> list[AppDBTileRecord]:
+        tile_records = []
+        num_tiles = len(chunk.Tiles)
+
+        for tile_no in range(num_tiles):
+            tile = chunk.Tiles[tile_no]
+
+            if tile.UniqId:
+                tile_ts = None
+                if ts := tile.TS:
+                    tile_ts = wintimestamp(ts)
+                tile_expiry_ts = None
+                if ts := tile.TS_Expiry:
+                    tile_expiry_ts = wintimestamp(ts)
+                name = tile.Name.strip("\x00")
+
+                xml_size = tile.ContentLength
+                tile_xml = chunk.TileXml[tile_no][:xml_size]
+                tile_xml = tile_xml.decode("utf-8", errors="replace")
```

Comment:
I thought we agreed on using errors="surrogateescape" ? If there is a special reason not do so, maybe we could add a comment?